app/src/main/java/in/ac/bits_pilani/goa/ard/utils/Browser.java

```diff
@@ -0,0 +1,370 @@
+    }
+
+    /**
+     * Disable action button in toolbar.
+     * @return current instance.
+     */
+    public Browser disableActionButton() {
+        actionButtonEnabled = false;
+        return this;
+    }
+
+    /**
+     * Sets the action button that is displayed in the Toolbar.
+     * @param icon The icon.
+     * @param description The description for the button. To be used for accessibility.
+     * @param pendingIntent pending intent delivered when the button is clicked.
+     * @param shouldTint Whether the action button should be tinted.
+     * @return current instance.
+     */
+    public Browser setActionButton(@NonNull final Bitmap icon,
+                                   @NonNull final String description,
+                                   @NonNull final PendingIntent pendingIntent,
+                                   final boolean shouldTint) {
+        actionButtonEnabled = true;
+        actionButton = new Bundle();
+        actionButton.putParcelable(KEY_ACTION_BUTTON_ICON, icon);
+        actionButton.putParcelable(KEY_ACTION_BUTTON_INTENT, pendingIntent);
+        actionButton.putString(KET_ACTION_BUTTON_DESC, description);
+        actionButton.putBoolean(KEY_ACTION_BUTTON_TINT, shouldTint);
+        return this;
+    }
+
+    /**
+     * Launche url in customized tab.
+     * @param url string url.
+     */
+    public void launchUrl(final String url) {
+        final String saneUrl = URLUtil.guessUrl(url);
+        if (URLUtil.isNetworkUrl(saneUrl)) {
+            initiateIntent();
+            customTabsIntent.launchUrl(hostActivity, Uri.parse(saneUrl));
+        } else {
+            hostActivity.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    Toast.makeText(hostActivity, "Invalid URL", Toast.LENGTH_SHORT).show();
+                }
+            });
+        }
+    }
+
+    /**
+     * Initiate customTabsIntent.
+     */
+    private void initiateIntent() {
+        final CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
+
+        builder.setToolbarColor(toolBarColor);
+        builder.setCloseButtonIcon(closeButtonIcon);
+        builder.setStartAnimations(hostActivity, startAnimEnterResId, startAnimExitResId);
+        builder.setExitAnimations(hostActivity, exitAnimEnterResId, exitAnimExitResId);
+        if (urlBarHiding) {
+            builder.enableUrlBarHiding();
+        }
+        builder.setShowTitle(titleEnabled);
+        if (menuItems != null) {
+            for (final Bundle item
+                    : menuItems) {
+                //noinspection ConstantConditions
+                builder.addMenuItem(item.getString(KEY_MENU_ITEM_TITLE),
+                        (PendingIntent) item.getParcelable(KEY_PENDING_INTENT));
+            }
+        }
+        if (shareInMenu) {
+            builder.addDefaultShareMenuItem();
+        }
+        if (actionButtonEnabled) {
+            //noinspection ConstantConditions
```

Comment:
remove noinspection.
and use some default value when you use bundle.getString(key, default);
same applies to all other areas where default values have not been used